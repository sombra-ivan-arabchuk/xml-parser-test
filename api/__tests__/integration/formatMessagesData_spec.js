const { formatMessagesData } = require("../../helpers/dataFormatters");
const readJsonFromXml = require("../../utils/xmlReader").readJsonFromXML;
const { messages } = require("../../utils/logMessages");

describe("formatMessagesData()", () => {
  it("check for data that will be represented for response", async () => {
    const jsonData = await readJsonFromXml("/api/CodeTest-XML.xml");
    expect(formatMessagesData(jsonData)).toMatchSnapshot();
  });

  it("check error case", async () => {
    // let's specify incorrect path and check the response
    try {
      await readJsonFromXml("/api/CodeTest-1XML.xml");
    } catch (e) {
      expect(e.message).toEqual(messages.readFileError);
    }
  });
});
