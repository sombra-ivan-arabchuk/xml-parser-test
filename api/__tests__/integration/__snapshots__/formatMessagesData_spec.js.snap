// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatMessagesData() check for data that will be represented for response 1`] = `
Array [
  Object {
    "content": "Hi Jane",
    "username": "Joe.doe@gmail.com",
  },
  Object {
    "content": "Hi Jane. How was yur day today. e are not trading",
    "username": "JANE.DOE@gmail.com",
  },
  Object {
    "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi",
    "username": "Joe.doe@gmail.com",
  },
  Object {
    "content": "Great to hear. Neque porro quisquam est qui dolorem ipsum quia dolor sit amet,consectetur, adipisci velit...",
    "username": "JANE.DOE@gmail.com",
  },
]
`;

exports[`getMessageContent() check for mapping of different objects 1`] = `
Array [
  Object {
    "content": "Hi Jane",
    "username": "Joe.doe@gmail.com",
  },
  Object {
    "content": "Hi Jane. How was yur day today. e are not trading",
    "username": "JANE.DOE@gmail.com",
  },
  Object {
    "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi",
    "username": "Joe.doe@gmail.com",
  },
  Object {
    "content": "Great to hear. Neque porro quisquam est qui dolorem ipsum quia dolor sit amet,consectetur, adipisci velit...",
    "username": "JANE.DOE@gmail.com",
  },
]
`;
